ui/pages/register.tsx

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable */
+
+import { RegistrationFlow, UpdateRegistrationFlowBody } from "@ory/client";
+import { Spinner } from "@canonical/react-components";
+import { AxiosError } from "axios";
+import type { NextPage } from "next";
+import { useRouter } from "next/router";
+import React, { useCallback, useEffect, useState } from "react";
+import { handleFlowError } from "../util/handleFlowError";
+import { Flow } from "../components/Flow";
+import { kratos } from "../api/kratos";
+import PageLayout from "../components/PageLayout";
+import {
+  UpdateRegistrationFlowWithOidcMethod,
+  UpdateRegistrationFlowWithPasswordMethod,
+  UpdateRegistrationFlowWithProfileMethod,
+} from "@ory/client/api";
+
+type FlowPreparer = (values: any) => UpdateRegistrationFlowBody;
+type SupportedFlowMethods = "oidc" | "password" | "profile";
+type PreparerMap = {
+  [method in SupportedFlowMethods]: FlowPreparer;
+};
+
+interface Traits {
+  email?: string;
+  full_name?: string;
+}
+
+function buildTraits(values: any): Traits {
+  return {
+    email: values["traits.email"]
+      ? (values["traits.email"] as string)
+      : undefined,
+    full_name: values["traits.full_name"]
+      ? (values["traits.full_name"] as string)
+      : undefined,
+  };
+}
+
+const flowPreparerMap: PreparerMap = {
+  oidc: (values: any): UpdateRegistrationFlowBody => ({
+    ...(values as UpdateRegistrationFlowWithOidcMethod),
+    method: "oidc",
+  }),
+  password: (values: any): UpdateRegistrationFlowBody => ({
+    ...(values as UpdateRegistrationFlowWithPasswordMethod),
+    method: "password",
+    traits: buildTraits(values),
+  }),
+  profile: (values: any): UpdateRegistrationFlowBody => ({
+    ...(values as UpdateRegistrationFlowWithProfileMethod),
+    method: "profile",
+    traits: buildTraits(values),
+  }),
+};
+
+function getFlowMethod(values: any): SupportedFlowMethods {
```

Comment:
question: are we supposed to support sign up with passwordless method after all? If not we should filter out the webauthn nodes from ui (let's at least leave a comment for the web team).
Otherwise we get `Failed to parse registration flow` error after an attempt to register with webauthn.

Reply:
Also this would be for the frontend team which will have this PR and comments along with my dummy code as a reference.